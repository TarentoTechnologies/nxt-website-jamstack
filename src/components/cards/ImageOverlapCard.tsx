import { GatsbyImage, getImage } from "gatsby-plugin-image";
import React from "react";
import ReactHtmlParser from "react-html-parser";

import {
  black60,
  black87,
} from "../../styles/style-guide/ColorGuide.module.css";
import { infoSectionH3 } from "../../styles/style-guide/Typography.module.css";
import {
  customBorderRadius,
  customPosLeft,
  customPosRight,
} from "./ImageOverlapCard.module.css";

interface ImageOverlapCardProps {
  img: any;
  content: any;
  subTitle: string;
  textPos?: string;
}

export const ImageOverlapCard = ({
  img,
  textPos,
  content,
  subTitle,
}: ImageOverlapCardProps) => {
  const image: any = getImage(img?.localFile);

  return (
    <div className="" data-testid="image-overlap-card">
      <div className="row position-relative my-5">
        <div
          className={`col-lg-6 ${textPos === "right" ? "order-2" : "order-1"}`}
        >
          <GatsbyImage
            image={image}
            alt={img ? img?.alternativeText : ""}
            className={`img-fluid`}
            data-aos="fade-up"
          />
        </div>
        <div
          className={`col-lg-6 ${customBorderRadius} ${
            textPos === "right" ? "order-1" : "order-2"
          } mb-5 mb-sm-5 mb-md-5 mb-lg-0`}
        >
          <div
            className={`${
              textPos === "right" ? customPosRight : customPosLeft
            } card layoutBg border border-0 p-4`}
          >
            <h4 className={`${black87} pb-2 ${infoSectionH3}`}>{subTitle}</h4>
            <p className={`${black60}`}>
              {" "}
              {ReactHtmlParser(content?.data?.childMarkdownRemark?.html)}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};
