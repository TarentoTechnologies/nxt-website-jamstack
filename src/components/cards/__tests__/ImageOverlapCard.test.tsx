import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { ImageOverlapCard } from "../ImageOverlapCard";

const imgData = {
  alternativeText: "Person designing a sitemap for a website",
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAKABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAQCAwX/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAAB1VrEzWSIF//EABwQAQABBAMAAAAAAAAAAAAAAAEAAgMEESEyM//aAAgBAQABBQI0F1CHBMjyp6//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQIBAT8BdJ//xAAYEAADAQEAAAAAAAAAAAAAAAAAASEQEf/aAAgBAQAGPwKC6oyYxH//xAAYEAADAQEAAAAAAAAAAAAAAAAAARExof/aAAgBAQABPyHOJDeKkoqlgeD8jmP/2gAMAwEAAgADAAAAEKDP/8QAFhEAAwAAAAAAAAAAAAAAAAAAEBFB/9oACAEDAQE/EFR//8QAFxEBAQEBAAAAAAAAAAAAAAAAAQARcf/aAAgBAgEBPxA1wlv/xAAcEAEBAQABBQAAAAAAAAAAAAABEQAhMVFhkbH/2gAIAQEAAT8QERtWGVIBJjXx03fEcm+czZeKPeaoqqPm/9k=",
        },
        images: {
          fallback: {
            src: "/static/384eeb371f7305cd9cd8e26f391b4cd7/a6312/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png",
            srcSet:
              "/static/384eeb371f7305cd9cd8e26f391b4cd7/e20c6/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 300w,\n/static/384eeb371f7305cd9cd8e26f391b4cd7/62ea7/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 600w,\n/static/384eeb371f7305cd9cd8e26f391b4cd7/a6312/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 1200w",
            sizes: "(min-width: 1200px) 1200px, 100vw",
          },
          sources: [],
        },
        width: 1200,
        height: 600,
      },
    },
  },
};

const desc = {
  data: {
    id: "9b078a46-9cae-558a-90d3-8a33fe2878f9",
    childMarkdownRemark: {
      html: "<p>From The Scratch - To build new experiences ground-up. we take utmost care to dive deep into problem space by employing first principles to figure out the fundamental truths that goes ignored otherwise</p>",
    },
  },
};

describe("Test if the ImageOverlapCard component is working as expected", () => {
  it("<ImageOverlapCard /> matches snapshot", () => {
    const component = render(
      <ImageOverlapCard
        img={imgData}
        content={desc}
        subTitle={"ALPHA Sprint i.e. Build From the Scratch"}
        textPos="right"
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<ImageOverlapCard /> renders without exploding", () => {
    const component = render(
      <ImageOverlapCard
        img={imgData}
        content={desc}
        subTitle={"ALPHA Sprint i.e. Build From the Scratch"}
        textPos="right"
      />
    );
    expect(component.getByTestId("image-overlap-card")).toBeInTheDocument();
  });
});
