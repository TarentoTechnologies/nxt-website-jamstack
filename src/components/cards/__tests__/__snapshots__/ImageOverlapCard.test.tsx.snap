// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the ImageOverlapCard component is working as expected <ImageOverlapCard /> matches snapshot 1`] = `
<div>
  <div
    class=""
    data-testid="image-overlap-card"
  >
    <div
      class="row position-relative my-5"
    >
      <div
        class="col-lg-6 order-2"
      >
        <div
          class="gatsby-image-wrapper gatsby-image-wrapper-constrained img-fluid"
          data-gatsby-image-wrapper=""
          style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
        >
          <div
            style="max-width: 1200px; display: block;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='600'%20width='1200'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
              style="max-width: 100%; display: block; position: static;"
            />
          </div>
          <img
            alt=""
            aria-hidden="true"
            data-placeholder-image=""
            decoding="async"
            src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAKABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAQCAwX/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAAB1VrEzWSIF//EABwQAQABBAMAAAAAAAAAAAAAAAEAAgMEESEyM//aAAgBAQABBQI0F1CHBMjyp6//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQIBAT8BdJ//xAAYEAADAQEAAAAAAAAAAAAAAAAAASEQEf/aAAgBAQAGPwKC6oyYxH//xAAYEAADAQEAAAAAAAAAAAAAAAAAARExof/aAAgBAQABPyHOJDeKkoqlgeD8jmP/2gAMAwEAAgADAAAAEKDP/8QAFhEAAwAAAAAAAAAAAAAAAAAAEBFB/9oACAEDAQE/EFR//8QAFxEBAQEBAAAAAAAAAAAAAAAAAQARcf/aAAgBAgEBPxA1wlv/xAAcEAEBAQABBQAAAAAAAAAAAAABEQAhMVFhkbH/2gAIAQEAAT8QERtWGVIBJjXx03fEcm+czZeKPeaoqqPm/9k="
            style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          />
          <img
            alt="Person designing a sitemap for a website"
            data-aos="fade-up"
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="/static/384eeb371f7305cd9cd8e26f391b4cd7/a6312/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png"
            data-srcset="/static/384eeb371f7305cd9cd8e26f391b4cd7/e20c6/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 300w,/static/384eeb371f7305cd9cd8e26f391b4cd7/62ea7/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 600w,/static/384eeb371f7305cd9cd8e26f391b4cd7/a6312/Person_designing_a_sitemap_for_a_website_redesign_c5a8151767.png 1200w"
            decoding="async"
            loading="lazy"
            sizes="(min-width: 1200px) 1200px, 100vw"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
          <noscript />
          <script
            type="module"
          >
            const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
          </script>
        </div>
      </div>
      <div
        class="col-lg-6 customBorderRadius order-1 mb-5 mb-sm-5 mb-md-5 mb-lg-0"
      >
        <div
          class="customPosRight card layoutBg border border-0 p-4"
        >
          <h4
            class="black87 pb-2 infoSectionH3"
          >
            ALPHA Sprint i.e. Build From the Scratch
          </h4>
          <p
            class="black60"
          >
             
            <p>
              From The Scratch - To build new experiences ground-up. we take utmost care to dive deep into problem space by employing first principles to figure out the fundamental truths that goes ignored otherwise
            </p>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
